akka-actor/src/main/scala/akka/japi/function/Function.scala

```diff
@@ -15,6 +15,18 @@ trait Function[-T, +R] extends java.io.Serializable {
   def apply(param: T): R
 }
 
+object Function extends java.io.Serializable {
+
+  private final val constFunc: Function[Any, Any] = new Function[Any, Any] {
+    override def apply(param: Any): Any = param
+  }
+
+  /**
+   * A method that returns its input value.
+   */
+  def identity[T](): Function[T, T] = constFunc.asInstanceOf[Function[T, T]]
+}
```

Comment:
I guess this will only work when we have `akka.japi.function.Function` as the type, not with `java.util.function.Function`. Might be confusing for users that it can be used somewhere but not everywhere.

What is the motivation for adding this? Is it a performance improvement over writing `x -> x`?

Reply:
The main reason is to provide consistency API experience, cause we are not using the `java.util.function.Function` but `akka.japi.function.Function`. the 2nd is performance reason.

Or we can add the overloaded methods for `java.util.function.*` and deprecated these which accept `akka.japi.function.*`?

I have to say when programming with `akka.japi.function`, there do have some confusion. I suggest we migrate the `Function0`,`Function1` and `Function2` to `java.util.function.*` later.

Reply:
> The main reason is to provide consistency API experience

I think this proposal is quite the opposite. `identity` can be used somewhere but not everywhere since we use a mix of akka.japi.function and java.util.function.

For the user it is better consistency to learn to use `x -> x`.